modules/cp4ba/main.tf

```diff
@@ -11,79 +7,117 @@ locals {
+    COMMON_SERVICE_FILE_sha1          = sha1(local.common_service_file_content)
   }
 
   provisioner "local-exec" {
     command = "${path.module}/scripts/install_cp4ba.sh"
 
+
     environment = {
       # ---- Cluster ----
       KUBECONFIG                    = var.cluster_config_path
-
       # ---- Platform ----
       CP4BA_PROJECT_NAME            = var.cp4ba_project_name
-
       # ---- Registry Images ----
-      ENTITLED_REGISTRY_EMAIL       = var.entitled_registry_user
-      ENTITLED_REGISTRY_KEY         = var.entitlement_key
+      ENTITLED_REGISTRY_EMAIL       = var.entitled_registry_user_email
+      ENTITLED_REGISTRY_KEY         = var.entitled_registry_key
       DOCKER_SERVER                 = local.docker_server
       DOCKER_USERNAME               = local.docker_username
-
       # ------- FILES ASSIGNMENTS --------
-      OPERATOR_PVC_FILE                = local.pvc_file
-      OPERATOR_GROUP_FILE              = local.operator_group_file
+      OPERATOR_SHARED_PV_CONTENT       = local.operator_shared_pv_file_content
+      SHARED_LOG_PV_CONTENT            = local.shared_log_pv_file_content
+      OPERATOR_SHARED_PVC_CONTENT      = local.operator_shared_pvc_file_content
+      SHARED_LOG_PVC_CONTENT           = local.shared_log_pvc_file_content
+      OPERATOR_GROUP_CONTENT           = local.operator_group_file_content
       CATALOG_SOURCE_FILE              = local.catalog_source_file
       COMMON_SERVICE_FILE              = local.common_service_file
-      CP4BA_SUBSCRIPTION_FILE          = local.cp4ba_subscription_file
-      CP4BA_DEPLOYMENT_CREDENTIALS_FILE = local.cp4ba_deployment_credentials_file
+      CP4BA_SUBSCRIPTION_CONTENT       = local.cp4ba_subscription_file_content
+      CP4BA_DEPLOYMENT_CREDENTIALS_CONTENT = local.cp4ba_deployment_credentials_file_content
       CP4BA_DEPLOYMENT_CONTENT         = local.cp4ba_deployment_file_content
       SECRETS_CONTENT                  = local.secrets_content
       ROLES_FILE                       = local.roles_file
       ROLE_BINDING_FILE                = local.role_binding_file
+
+      ldap_admin              = var.ldap_admin_name
+      ldap_password           = var.ldap_admin_password
+      ldap_host_ip            = var.ldap_host_ip
+      # ----- DB2 Settings -----
+      db2_host_port           = var.db2_ports
+      db2_host_address        = var.db2_host_address
+      db2_admin               = var.db2_admin_username
+      db2_password            = var.db2_admin_user_password
+      db2_user                = var.db2_user
+    }
+  }
+
+  provisioner "local-exec" {
+    when        = destroy
+    command     = "./uninstall_cp4ba.sh"
+    working_dir = "${path.module}/scripts"
+
+    environment = {
+      kubeconfig       = self.triggers.kubeconfig
+      cp4ba_project_name = self.triggers.cp4ba_project_name
     }
   }
 }
 
 data "external" "get_endpoints" {
-  count = var.enable ? 1 : 0
+  count = 1
```

Comment:
Since you are using the `var.enable` in the `install_cp4ba` module, then this should still be `count = var.enable ? 1 : 0`

Reply:
Fixed.
`count = var.enable_cp4ba ? 1 : 0`